src/pymovements/utils/archives.py

```diff
@@ -151,10 +151,16 @@ def _extract_tar(
         Compression filename suffix.
     """
     with tarfile.open(source_path, f'r:{compression[1:]}' if compression else 'r') as archive:
-        if sys.version_info < (3, 12):  # pragma: <3.12 cover
-            archive.extractall(destination_path)
-        else:  # pragma: >=3.12 cover
-            archive.extractall(destination_path, filter='tar')
+        for member in archive.getnames():
+            if (
+                os.path.exists(os.path.join(destination_path, member)) and
```

Comment:
only checking if the path exists can be an issue if the user aborted extraction. then there will be a partially written file on the system and this if condition won't notice that.

luckily the [`TarInfo`](https://docs.python.org/3/library/tarfile.html#tarfile.TarInfo) has fields for `size` and `chksum`.

Checking the size should be sufficient for most use cases.
You can create a private function in `archives.py` to make the if condition a bit more readable.

For paranoid users we could add checksum functionality after extraction in another PR.

Reply:
any updates on checking the file size?

Reply:
there is a problem when we have `remove_top_level`, let's discuss Wednesday

Reply:
Until then, can you elaborate more on the problem in a written comment?
For me it's easier to process and decide when it's written rather than spoken.

Reply:
basically, if you extract twice, and `remove_finished` or `remove_top_level` double extraction does not work -- since the original extraction is removed. 

but this is a user choice and one can expect if you delete the download that you can't extract it anymore